main.nf

```diff
@@ -0,0 +1,684 @@
+ * Visualization
+ ************************************************************/
+
+/*
+ * Fusion Inspector
+ */
+process fusion_inspector {
+    tag "$name"
+    publishDir "${params.outdir}/tools/FusionInspector", mode: 'copy'
+
+    when:
+    (params.fusion_inspector && !params.singleEnd) || (params.fusion_inspector && params.test)
+
+    input:
+    set val(name), file(reads) from read_files_fusion_inspector
+    file reference from fusion_inspector_ref.collect()
+    file summary_fusions
+
+    output:
+    file '*' into fusion_inspector_output
+
+    script:
+    """
+    FusionInspector \\
+        --fusions ${summary_fusions} \\
+        --genome_lib ${reference} \\
+        --left_fq ${reads[0]} \\
+        --right_fq ${reads[1]} \\
+        --CPU ${task.cpus} \\
+        --out_dir . \\
+        --out_prefix finspector \\
+        --prep_for_IGV
+    """
+}
+
+/*************************************************************
+ * Building report
+ ************************************************************/
+
+/*
+ * Parse software version numbers
+ */
+process get_software_versions {
+
+    when:
+    !params.test
+
+    output:
+    file 'software_versions_mqc.yaml' into software_versions_yaml
+
+    script:
+    """
+    echo $workflow.manifest.version > v_pipeline.txt
+    echo $workflow.nextflow.version > v_nextflow.txt
+    fastqc --version > v_fastqc.txt
+    multiqc --version > v_multiqc.txt
+    cat $baseDir/tools/fusioncatcher/Dockerfile | grep "VERSION" > v_fusioncatcher.txt
+    cat $baseDir/tools/fusion-inspector/environment.yml | grep "fusion-inspector" > v_fusion_inspector.txt
+    cat $baseDir/tools/star-fusion/environment.yml | grep "star-fusion" > v_star_fusion.txt
+    cat $baseDir/tools/ericscript/environment.yml | grep "ericscript" > v_ericscript.txt
+    cat $baseDir/tools/pizzly/environment.yml | grep "pizzly" > v_pizzly.txt
+    cat $baseDir/tools/squid/environment.yml | grep "squid" > v_squid.txt
+    scrape_software_versions.py > software_versions_mqc.yaml
+    """
+}
+
+/*
+ * FastQC
+ */
+process fastqc {
+    tag "$name"
+    publishDir "${params.outdir}/fastqc", mode: 'copy',
+        saveAs: {filename -> filename.indexOf(".zip") > 0 ? "zips/$filename" : "$filename"}
+
+    when:
+    !params.test
+
+    input:
+    set val(name), file(reads) from read_files_fastqc
+
+    output:
+    file "*_fastqc.{zip,html}" into fastqc_results
+
+    script:
+    """
+    fastqc -q $reads
+    """
+}
+
+/*
+ * MultiQC
+ */
+process multiqc {
+    tag "$name"
+    publishDir "${params.outdir}/MultiQC", mode: 'copy'
+
+    when:
+    !params.test
+
+    input:
+    set val(name), file(reads) from read_files_multiqc
+    file multiqc_config
+    file ('fastqc/*') from fastqc_results.collect()
+    file ('software_versions/*') from software_versions_yaml
+    file workflow_summary from create_workflow_summary(summary)
+    file fusions_mq from summary_fusions_mq.ifEmpty('')
```

Comment:
I like to put these in subdirectories just in case there are filename clashes (especially when using `*` as output channels 😉 )

Also, I think you need `.collect()` again for multiple samples. The syntax I usually use is `from channelname.collect().ifEmpty([])`
```suggestion
    file ('fusions_mq/*') from summary_fusions_mq.collect().ifEmpty([])`
```

Reply:
Also, it's a shame that there aren't more MultiQC modules available for these tools! It would be good to write some 😉

Reply:
![future](https://user-images.githubusercontent.com/3359239/51391756-3a7c0880-1b33-11e9-8865-32ee07a0256e.png)